value/src/test/java/com/google/auto/value/processor/ExtensionTest.java

```diff
@@ -0,0 +1,265 @@
+package com.google.auto.value.processor;
+
+import com.google.auto.value.AutoValueExtension;
+import com.google.testing.compile.JavaFileObjects;
+import junit.framework.TestCase;
+
+import javax.lang.model.element.ExecutableElement;
+import javax.tools.JavaFileObject;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.google.common.truth.Truth.assertAbout;
+import static com.google.testing.compile.JavaSourceSubjectFactory.javaSource;
+
+/**
+ * Created by rharter on 5/5/15.
+ */
+public class ExtensionTest extends TestCase {
+  public void testExtensionCompilation() throws Exception {
+
+    JavaFileObject javaFileObject = JavaFileObjects.forSourceLines(
+        "foo.bar.Baz",
+        "package foo.bar;",
+        "",
+        "import com.google.auto.value.AutoValue;",
+        "",
+        "@AutoValue",
+        "public abstract class Baz {",
+        "  abstract String foo();",
+        "}");
+    JavaFileObject expectedOutput = JavaFileObjects.forSourceLines(
```

Comment:
I think I'd prefer not to have any new expected-output strings for the generated AutoValue implementation. They are a pain to maintain during changes. I think it would be enough just to assert on expectedExtensionOutput.